src/GafferCycles/IECoreCyclesPreview/SocketAlgo.cpp

```diff
@@ -525,6 +538,101 @@ void setRampSocket( ccl::Node *node, const ccl::SocketType *socket, const IECore
 	node->set( *socket, ramp );
 }
 
+ccl::ParamValue setParamValue( const ccl::ustring &name, const IECore::Data *value )
+{
+	switch( value->typeId() )
+	{
+		case BoolDataTypeId :
+		case IntDataTypeId :
+		case UIntDataTypeId :
+		case DoubleDataTypeId :
+			if( const FloatData *data = runTimeCast<const FloatData>( value ) )
+			{
+				const float result = data->readable();
+				return ccl::ParamValue( name, ccl::TypeDesc::TypeFloat, 1, &result );
+			}
+			break;
+		case FloatDataTypeId :
+			if( const FloatData *data = dataCast<const FloatData>( name.c_str(), value ) )
+			{
+				const float result = data->readable();
+				return ccl::ParamValue( name, ccl::TypeDesc::TypeFloat, 1, &result );
+			}
+			break;
+		case Color3fDataTypeId :
+			if( const Color3fData *data = dataCast<const Color3fData>( name.c_str(), value ) )
+			{
+				const ccl::float3 result = setColor( data->readable() );
```

Comment:
As I understand it, `ParamValue()` just wants a pointer to 3 floats here, so I don't see any benefit to converting to `ccl::float3`. Can we just pass `data->readable().getValue()` direct to `ParamValue`? I think this applies to many of the other cases in this switch.

Reply:
Sounds good, I didn't really see/know of that call when browsing the headers.